### PR TITLE
Add Course  API Endpoints to Home Page

### PR DESCRIPTION
**Pull request by adarsh-technocrat**
**Describe the changes you've made**

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**

**Additional context (OPTIONAL)**

**Test plan (OPTIONAL)**

A good test plan should give instructions that someone else can easily follow.

**Checklist**

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**
Preview: Deploy preview link here with the appropriate route

<!-- Please add deployed link to here text >

### CODE REVIEW

File: src/services/courseServices.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const BASE_URL = "https://jsonplaceholder.typicode.com";
+
+export const fetchCourseData = () => {
+  return async (dispatch) => {
+    try {
+      dispatch(featchCourseRequest());
+      const response = await axios.get(`${BASE_URL}/posts`); // dummy URL for testing API call
+      const data = await response.data;
+      dispatch(featchCourseSuccess(data));
```
Comment (Abhishek-kumar09) [inline src/services/courseServices.js:13]: Why are you dispatching `courseRequestFetch` twice?

Comment (Abhishek-kumar09) [reply src/services/courseServices.js:13]: Please also correct spelling of `fetch`

Comment (adarsh-technocrat) [reply src/services/courseServices.js:13]: > Please also correct spelling of `fetch`

Sure @Abhishek-kumar09  I'll resolve that typo!

Comment (adarsh-technocrat) [reply src/services/courseServices.js:13]: > Why are you dispatching `courseRequestFetch` twice?

@Abhishek-kumar09   `fetchCourseRequest()` action is dispatch to set loading state `ture` once the user gets the Data  than another  `fetchCourseSuccess(data)` action is dispatch which set `loading` to  `false` and  store response data to `course:[ ]`.

As there are two functions one for  `course/all` and `course/:id`  in both the functions I have used the same `actions` for handling the states.

Comment (Abhishek-kumar09) [reply src/services/courseServices.js:13]: Other things looks good to me.

**Comment by adarsh-technocrat** [comment]
Hey @Abhishek-kumar09  PR is up for Review :)

File: src/components/CourseMediaCard/MediaCard.js [hunk 2]
```diff
@@ -67,32 +66,35 @@ function MediaCard(props) {
...
-              <Typography variant="subtitle2">{description} </Typography>
+              <Typography variant="subtitle2">{courseShortDescription} </Typography>
             </Box>
           </Box>
         </CardContent>
       </CardActionArea>
       <CardActions className={classes.cardFooter}>
         <StarIcon className={classes.starIcon} />
-        <Typography>{ratings}</Typography>
+        <Typography>{raiting}</Typography>
```
Comment (Abhishek-kumar09) [inline src/components/CourseMediaCard/MediaCard.js:89]: typo

File: src/MockServer/courses/courseDetail/index.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
+export const courseDetail = {
+  courseID: "1",
+  tag: {
+    tagId: "1254",
+    tagName: "web development",
+  },
```
Comment (Abhishek-kumar09) [inline src/MockServer/courses/courseDetail/index.js:6]: it can be 
```
tag: ["Web Development", "React"]
```

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

**Pull request merged**